step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -318,11 +314,22 @@ protected Staging createStaging(){
     protected void fillStaging(Staging staging, AutomationPackageContent packageContent, AutomationPackage newPackage, AutomationPackage oldPackage, ObjectEnricher enricherForIncludedEntities, AutomationPackageArchive automationPackageArchive){
         staging.plans = preparePlansStaging(packageContent, oldPackage, enricherForIncludedEntities);
         staging.taskParameters = prepareExecutionTasksParamsStaging(enricherForIncludedEntities, packageContent, oldPackage, staging.plans);
-        staging.functions = prepareFunctionsStaging(newPackage, automationPackageArchive, packageContent, enricherForIncludedEntities, oldPackage);
+        staging.functions = prepareFunctionsStaging(newPackage, automationPackageArchive, packageContent, enricherForIncludedEntities, oldPackage, staging.resourceManager);
     }
 
     protected void persistStagedEntities(Staging staging,
                                          ObjectEnricher objectEnricher) {
+        List<Resource> stagingResources = staging.resourceManager.findManyByCriteria(null);
+        try {
+            for (Resource resource: stagingResources) {
+                resourceManager.copyResource(resource, staging.resourceManager);
+            }
+        } catch (IOException | SimilarResourceExistingException | InvalidResourceFormatException e) {
+            throw new AutomationPackageManagerException("Unable to persist a resource in automation package", e);
+        } finally {
+            staging.resourceManager.cleanup();
```

Comment:
Not sure if it is the best place to cleanup the staging resource manager but it does the job for the moment